packages/auth/src/grant/routes.ts

```diff
@@ -253,3 +255,76 @@ async function continueGrant(
     }
   }
 }
+
+// eslint-disable @typescript-eslint/no-explicit-any
+function isAccessTokenUpdateParameter(
+  param: any
+): param is {
+  access: AccessRequest[]
+} {
+  return (
+    param &&
+    param.access &&
+    Array.isArray(param.access) &&
+    param.access.length > 0 &&
+    param.access.every((a: any) => isAccessRequest(a))
+  )
+}
+// eslint-enable
+
+async function updateGrant(
+  deps: ServiceDependencies,
+  ctx: AppContext
+): Promise<void> {
+  // TODO: httpsig validation
+  const { continueId } = ctx.params
+  const continueToken = (ctx.headers['authorization'] as string)?.split(
+    'GNAP '
+  )[1]
+  const {
+    interact_ref: interactRef,
+    access_token: accessToken
+  } = ctx.request.body
+
+  if (
+    !continueId ||
+    !continueToken ||
+    !interactRef ||
+    !isAccessTokenUpdateParameter(accessToken)
+  ) {
+    ctx.status = 401
+    ctx.body = {
+      error: 'invalid_request'
+    }
+  } else {
+    const { config, accessTokenService, grantService, accessService } = deps
+    const grant = await grantService.getByContinue(
+      continueId,
+      continueToken,
+      interactRef
+    )
+
+    if (grant) {
+      await grantService.update(grant.id, accessToken.access)
+      const updatedAccessToken = await accessTokenService.create(grant.id)
+      const updatedAccess = await accessService.getByGrant(grant.id)
+
+      // TODO: add "continue" to response if additional grant request steps are added
+      ctx.body = {
+        access_token: {
```

Comment:
We should add the `continue` field here if the level of access has changed, since we probably want the RO to re-authorize the new scope of access. [It looks like the continue token gets updated as well, between the example response to the initial POST and the example PATCH response.](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-5.3)

Reply:
Makes sense. I copied the `// TODO: add "continue" to response if additional grant request steps are added` part from another method. Does that still apply here in addition to "if the level of access has changed" or should continue be added _only_ if the level of access has changed?

Reply:
I think we should add continue either if there has been an access level change, or if the grant hasn't been issued yet. So it would be excluded if there's no access change and the grant has already been issued. In that second scenario you could include the `access_token`, I think.